wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -162,61 +167,130 @@ class CoinJoinMixingService @Inject constructor(
                 updateBlockChain(blockChain)
             }
             .launchIn(coroutineScope)
+
+        blockchainStateProvider.observeState()
+            .filterNotNull()
+            .distinctUntilChanged()
+            .onEach { blockChainState ->
+                log.info("coinjoin: new block: ${blockChainState.bestChainHeight}")
+                updateBalance(walletDataProvider.getWalletBalance())
+            }
+            .launchIn(coroutineScope)
+
+        walletDataProvider.observeBalance()
+            .onEach { balance ->
+                log.info("coinjoin: total balance: $balance (before distinct)")
+            }
+            //.distinctUntilChanged()
```

Comment:
These lines are for debugging.  

I may remove this observer as it doesn't work as well as I would like.  The `blockchainStateProvider.observeState()` may be sufficient.  CoinJoin works on confirmed transactions, which means that having an observer that is triggered every block is ideal.

Sometimes there are a lot of ANR's when running this PR -- maybe its related to this observer.  This observer doesn't seem to be triggered with each block, although it seems like it should.